catalog/core/catalog-core-attributeregistry/src/test/java/ddf/catalog/data/impl/AttributeRegistryImplTest.java

```diff
@@ -57,17 +91,56 @@ public void testRemoveAttribute() {
                 true,
                 false,
                 BasicTypes.STRING_TYPE);
-        assertThat(registry.register(descriptor), is(true));
+        registry.register(descriptor);
 
         Optional<AttributeDescriptor> descriptorOptional = registry.lookup(attributeName);
         assertThat(descriptorOptional.isPresent(), is(true));
 
-        registry.deregister(attributeName);
+        registry.deregister(descriptor);
         descriptorOptional = registry.lookup(attributeName);
         assertThat(descriptorOptional.isPresent(), is(false));
     }
 
     @Test
+    public void testRemoveDuplicateAttribute() {
+        final String attributeName = "foo";
+        final AttributeDescriptor descriptor1 = new AttributeDescriptorImpl(attributeName,
+                true,
+                true,
+                true,
+                true,
+                BasicTypes.STRING_TYPE);
+        registry.register(descriptor1);
+        registry.register(descriptor1);
+
+        registry.deregister(descriptor1);
+        final Optional<AttributeDescriptor> descriptorOptional = registry.lookup(attributeName);
+        assertThat(descriptorOptional.isPresent(), is(true));
+        assertThat(descriptorOptional.get(), is(descriptor1));
+
+        registry.deregister(descriptor1);
+        final Optional<AttributeDescriptor> descriptorOptional2 = registry.lookup(attributeName);
+        assertThat(descriptorOptional2.isPresent(), is(false));
+    }
+
+    @Test
+    public void testAddDuplicateAttribute() {
+        final String attributeName = "foo";
+        final AttributeDescriptor descriptor1 = new AttributeDescriptorImpl(attributeName,
+                true,
+                true,
+                true,
+                true,
+                BasicTypes.STRING_TYPE);
+        registry.register(descriptor1);
+        registry.register(descriptor1);
+
+        final Optional<AttributeDescriptor> descriptorOptional = registry.lookup(attributeName);
+        assertThat(descriptorOptional.isPresent(), is(true));
+        assertThat(descriptorOptional.get(), is(descriptor1));
+    }
+
+    @Test
     public void testAddAttributeWithSameName() {
```

Comment:
We may want a test that involves registering two different attributes with the same name, deregistering the first one, then verifying that subsequent calls to `lookup()` return the second one.